homeassistant/components/androidtv/__init__.py

```diff
@@ -114,7 +129,7 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     state_det_rules = entry.options.get(CONF_STATE_DETECTION_RULES)
     aftv, error_message = await async_connect_androidtv(
-        hass, entry.data, state_detection_rules=state_det_rules
+        hass, dict(entry.data), state_detection_rules=state_det_rules
```

Comment:
Why is the entry data copied?

Reply:
Just because the type `MappingProxy` is incompatible with the input function parameter type (`dict[str, Any]`).

Reply:
So maybe use `Mapping[str, Any]` as the typing on `async_connect_androidtv` for that parameter?

Reply:
But this method is also used by `config_flow` to test the connection based on `user_input`, and `user_input` is `dict[str, Any]`.

Reply:
Nevermind, type  `Mapping[str, Any]` is ok for both calls👍